app/components/forms/form-select.js

```diff
@@ -0,0 +1,49 @@
+import EmberPowerSelect from 'ember-power-select/components/power-select';
+import { computed } from '@ember/object';
+import { gte } from '@ember/object/computed';
+
+const OPTIONS_FOR_SEARCH = 5;
+
+const CSS_CLASSES = {
+  DISABLED: 'travis-form__field-component--disabled',
+  FIELD_COMPONENT: 'travis-form__field-component',
+  FIELD_SELECT: 'travis-form__field-select'
+};
+
+export default EmberPowerSelect.extend({
+  disabled: false,
+  placeholder: '',
+
+  searchEnabled: gte('options.length', OPTIONS_FOR_SEARCH),
+  searchPlaceholder: 'Type to filter options...',
+
+  allowClear: false,
+  horizontalPosition: 'auto',
+  verticalPosition: 'below',
+
+  triggerClass: computed('disabled', function () {
+    const classes = [CSS_CLASSES.FIELD_COMPONENT, CSS_CLASSES.FIELD_SELECT];
+    if (this.disabled) {
+      classes.push(CSS_CLASSES.DISABLED);
+    }
+    return classes.join(' ');
+  }),
+
+  onChange() {},
+  onFocus() {},
+  onBlur() {},
+  onInit() {},
+
+  onchange: computed(function () {
```

Comment:
Are there both `onChange` and `onchange` because this is inheriting from the addon component? Maybe there could be a comment to explain that, otherwise it’s pretty mysterious.

I haven’t thought deeply about this but is it possible this could be accomplished via composition rather than inheritance? If not, no big deal, it just worries me to be extending such a complex component, it seems like things could break more easily.

Reply:
1) You are right, `onchange` is the event handler exposed by the `ember-power-select` component and `onChange` is there mostly to align with other `travis-form` fields. Although, after I changed `travis-form` to expose `form-field` directly (which is responsible for exposing public events), it doesn't make sense. `form-field` can bind to `ember-power-select` native events internally. I'll change that, thanks for catching! 👍 

2) I also prefer composition over inheritance in regards with components (esp. external ones). But official add-on documentation suggest inheritance as a preferred way: https://ember-power-select.com/cookbook. I would suggest to keep it this way for now and see how it works. In case you have any issues with power select in future, please let me know.

Reply:
ah well if the documentation says to do it this way, I agree we should do it this way hehe thanks